java/plasma/src/main/java/org/apache/arrow/plasma/PlasmaClient.java

```diff
@@ -110,6 +109,33 @@ public void fetch(byte[][] objectIds) {
     PlasmaClientJNI.fetch(conn, objectIds);
   }
 
+  @Override
+  public List<ObjectStoreData> get(byte[][] objectIds, int timeoutMs) {
+    ByteBuffer[][] bufs = PlasmaClientJNI.get(conn, objectIds, timeoutMs);
+    assert bufs.length == objectIds.length;
+
+    List<ObjectStoreData> ret = new ArrayList<>();
+    for (int i = 0; i < bufs.length; i++) {
+      ByteBuffer databuf = bufs[i][0];
+      ByteBuffer metabuf = bufs[i][1];
+      if (databuf == null) {
+        ret.add(new ObjectStoreData(null, null));
+      } else {
+        byte[] data = new byte[databuf.remaining()];
+        databuf.get(data);
```

Comment:
[just for my understanding] is there a reason we are making a copy of the data? do we want them to immutable by the caller?

Reply:
In my understanding, the buffer is from Plasma shared memory. It is safe to make a copy and the user can release the shared buffer.  Add @pcmoritz to make sure whether my understanding is correct.

Reply:
I notice that the other `get` function has the same copy operation.